pkg/expectations/expectations.go

```diff
@@ -125,18 +128,18 @@ type ControllerExpectations struct {
 
 // GetExpectations returns the ControlleeExpectations of the given controller.
 func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
+	var err error
 	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
 		return exp.(*ControlleeExpectations), true, nil
-	} else {
-		return nil, false, err
 	}
+	return nil, false, err
```

Comment:
can just return `nil, false, nil`, right?

Reply:
@jeffvance Actually no, the (nil, false, err) case there is hit when there actual *is* an error so it's important.  The need to define the variable above is because I took out the explicit else to simplify the function, and in that case the go compiler gets upset because the def of the variable is in an if statement.

Anyway, not a lint problem; I can just put it back the way it was.  Or drop the compound statement and do a more traditional "if err != nil" pattern.

Reply:
I should have been clearer above. Here's what I was meaning to say:
```
func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
	var err error
	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
/// these 3 ^^ vars are scoped in the if block and not the func
		return exp.(*ControlleeExpectations), true, nil
	}
// here we have either 1) an err or 2) no err but !exists (but exists it not scoped here)
	return nil, false, err
// however, err in the return is always false (its "zero" value)
}
```
I think I would refactor as:
```
func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
	exp, exists, err := r.GetByKey(controllerKey)
        if exp != nil {
		return exp.(*ControlleeExpectations), exists, err
	}
	return nil, exists, err
}
```

Reply:
Much better, I'll add it, thanks!